hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/parsing/Util.java

```diff
@@ -106,6 +116,47 @@ public static String getMsgType(String hlMsg) {
 		}
 		return msgType;
 	}
+	
+	/**
+	 * returns the message id based on the HL7 message.
+	 * @param hlMsg
+	 * @return
+	 */
+	public static String getMsgId(String hlMsg) {
+
+		String msgId = "";
+
+		if (hlMsg == null || hlMsg.isEmpty()) {
```

Comment:
StringUtils.isEmpty(...) will handle both of these cases.

Reply:
done